src/api/app/datatables/project_datatable.rb

```diff
@@ -22,9 +24,11 @@ def projects
   # rubocop:disable Naming/AccessorMethodName
   def get_raw_records
     if projects
-      projects
+      projects.includes(label_globals: [:label_template_global]).references(:label_globals, :label_template_global)
     else
-      show_all ? Project.all : Project.filtered_for_list
+      # TODO: filtered_for_list scope causes multiple requests when eager loading causing errors when using the search:
+      # Mysql2::Error: Unknown column 'label_template_globals.name' in 'where clause'
+      show_all ? Project.all.includes(label_globals: [:label_template_global]).references(:label_globals, :label_template_global) : Project.filtered_for_list
```

Comment:
The regex match performed in the scope https://github.com/openSUSE/open-build-service/blob/e7eb2dfd7141b7648528264488a5bd8ba9a3b978/src/api/app/models/project.rb#L97 triggers more then one query when eager loading the associated labels which makes the search fail in the datatable.

Reply:
solved